bleak/backends/dotnet/client.py

```diff
@@ -333,13 +345,7 @@ def handler(sender, args):
                     return_type=DevicePairingResult,
                 )
 
-            try:
-                custom_pairing.PairingRequested -= handler
-            except Exception:
-                # TODO: Find a way to remove WinRT events...
-                pass
-            finally:
-                del handler
+            custom_pairing.remove_PairingRequested(pairing_requested_token)
```

Comment:
Should this be in a `finally:` to ensure that it is always removed even if an exception is raised between the time the event is added and here?

Reply:
Good idea. Do we have try and finally for other deregistrations of handlers? Or are they all safely removed on disconnect?

Reply:
I think most are handled in disconnect rather than short term like this.

Reply:
and I think we have something to remove the event handlers if the connection fails, but it wouldn't hurt to double-check

Reply:
Enclosed the removal in a finally clause just now.